modules/articles/client/config/articles.client.routes.js

```diff
@@ -12,22 +12,81 @@ angular.module('articles').config(['$stateProvider',
       })
       .state('articles.list', {
         url: '',
-        templateUrl: 'modules/articles/client/views/list-articles.client.view.html'
+        templateUrl: 'modules/articles/client/views/list-articles.client.view.html',
+        controller: 'ArticlesListController',
+        resolve:{
+          articles: function (Articles, $stateParams, $state) {
+            return Articles.query().$promise.then(
+              function (articles) {
+                //success
+                return articles;
+              },
+              function () {
+                //fail
+                $state.go('not-found');
+              }
+            );
+          }
+        }
       })
       .state('articles.create', {
         url: '/create',
         templateUrl: 'modules/articles/client/views/create-article.client.view.html',
+        controller: 'ArticlesController',
         data: {
           roles: ['user', 'admin']
+        },
+        resolve: {
+          article: function () {
+            return { mock: 'mock' };
+          }
         }
       })
       .state('articles.view', {
         url: '/:articleId',
-        templateUrl: 'modules/articles/client/views/view-article.client.view.html'
+        templateUrl: 'modules/articles/client/views/view-article.client.view.html',
+        controller: 'ArticlesController',
+        resolve: {
+          article: function (Articles, $stateParams, $state) {
+            return Articles.get({
```

Comment:
just return Articles.get(..);

the 404 should be handled by the http interceptor (https://github.com/meanjs/mean/pull/796/files#diff-90593c1684669b4a0cc185ec424f0a24R26)